schemas/questionnaire_v1.json

```diff
@@ -398,8 +398,25 @@
             ]
           }
         },
-        "required": ["id", "groups"]
+
+        "if": {
+          "properties": {
+            "questionnaire_flow": {
+              "properties": { "type": { "const": "Linear" } }
+            }
+          }
+        },
+        "then": {
+          "required": ["id", "groups"]
+        },
+        "else": {
+          "required": ["id", "groups", "title"]
+        }
```

Comment:
This doesn't seem to work. A Hub survey without a title is not passing for me.

Reply:
It's because `if` conditions keyword is not found in schema so `then` is being used, it might be impossible to do this logic in JSON schema definitions.

Reply:
It is possible. It is currently defined incorrect at the wrong level, JSON schema can't look upwards to a parent's value. So this need to set `"required": ["id", "groups"]` as this is always true.

Then at the top level you need:
```json
"allOf": [
    {
      "if": {
        "properties": {
          "questionnaire_flow": {
            "properties": {
              "type": {
                "const": "Hub"
              }
            }
          }
        }
      },
      "then": {
        "properties": {
          "sections": {
            "items": {
              "required": ["title"]
            }
          }
        }
      }
    }
  ]
```

Reply:
Thanks, I used it and it works like a charm!